site/src/components/AuditLogRow/AuditLogRow.tsx

```diff
@@ -0,0 +1,191 @@
+import Collapse from "@material-ui/core/Collapse"
+import { makeStyles } from "@material-ui/core/styles"
+import TableCell from "@material-ui/core/TableCell"
+import TableRow from "@material-ui/core/TableRow"
+import { AuditLog } from "api/typesGenerated"
+import { CloseDropdown, OpenDropdown } from "components/DropdownArrows/DropdownArrows"
+import { Pill } from "components/Pill/Pill"
+import { Stack } from "components/Stack/Stack"
+import { UserAvatar } from "components/UserAvatar/UserAvatar"
+import { t } from "i18next"
+import { ComponentProps, useState } from "react"
+import { MONOSPACE_FONT_FAMILY } from "theme/constants"
+import userAgentParser from "ua-parser-js"
+import { createDayString } from "util/createDayString"
+import { AuditLogDiff } from "./AuditLogDiff"
+
+const pillTypeByHttpStatus = (httpStatus: number): ComponentProps<typeof Pill>["type"] => {
+  if (httpStatus >= 300 && httpStatus < 500) {
+    return "warning"
+  }
+
+  if (httpStatus >= 500) {
+    return "error"
+  }
+
+  return "success"
+}
+
+const actionLabelByAction: Record<AuditLog["action"], string> = {
+  create: t("actions.create", { ns: "auditLog" }),
+  write: t("actions.write", { ns: "auditLog" }),
+  delete: t("actions.delete", { ns: "auditLog" }),
+}
+
+const resourceLabelByResourceType: Record<AuditLog["resource_type"], string> = {
+  organization: "organization",
+  template: "template",
+  template_version: "template version",
+  user: "user",
+  workspace: "workspace",
+}
+
+const readableActionMessage = (auditLog: AuditLog) => {
+  return `${actionLabelByAction[auditLog.action]} ${
+    resourceLabelByResourceType[auditLog.resource_type]
+  }`
```

Comment:
Here's how to do interpolation with react-i18n https://www.i18next.com/translation-function/interpolation

Reply:
We are planning to move this "message builder" to the BE so the CLI and FE can use the same language so I'm going just let this as it is but thanks for sharing the docs. Do you know how they could support interpolation with components?

Reply:
This is the closest I've found https://www.i18next.com/translation-function/formatting